### ISSUE TITLE
Wrong interface selected by PF_RING when using long interface names

### ISSUE THREAD

**Issue by jacobvw** [ntop/PF_RING#694]
We can reproduce this in our environment by having two interfaces named:
'test-10g-port1' and 'test-10g-port2'

After calling pfring_zc_open_device() the ifindex PF_RING binds to is checked with:
```
uint32_t ifindex = pfring_zc_get_queue_id(fdata->device);

fprintf(stderr, "interface %s pfring index %u ifindex %u\n", 
        interface, 
        QUEUEID_TO_IFINDEX(ifindex), 
        if_nametoindex(interface));
```

When using the interface 'test-10g-port2' PF_RING binds to the correct interface
`interface test-10g-port2 pfring index 31 ifindex 31`

When using the interface 'test-10g-port1' PF_RING binds to the incorrect interface
`interface test-10g-port1 pfring index 31 ifindex 30`

I believe PF_RING is ignoring the last character in the interface name and may be related to IFNAMSIZ.
This only occurs when using interface names >= 14 characters.

PF_RING ZC 7.9.0.210208
ubuntu 20.04
5.8.0-41-generic

**Comment by cardigliano** [comment]
Are you opening the queue using "test-10g-port1" or "zc:test-10g-port1"?

**Comment by cardigliano** [comment]
PF_RING uses the standard bind() call to bind the interface to the socket, this is using struct sockaddr which allows a 14-chars interface. In this case pf_ring should return an error rather than opening the wrong interface.

**Comment by cardigliano** [comment]
Fixed in f354d298418a1381addafb9a197313e98ef4b01d
